test/Npgsql.Tests/ReaderTests.cs

```diff
@@ -1488,9 +1488,11 @@ internal ExplodingTypeHandler(PostgresType postgresType, bool safe)
         public override int Read(NpgsqlReadBuffer buf, int len, FieldDescription? fieldDescription = null)
         {
             buf.ReadInt32();
-            throw _safe
-                ? new NpgsqlSafeReadException(new Exception("Safe read exception as requested"))
-                : throw new Exception("Non-safe read exception as requested");
+            if (!_safe)
+                buf.Connector.Break();
```

Comment:
Unless I'm mistaken, type handlers should never break - only the read buffer should do that because of I/O failure or the like. But it's OK to leave it like this, e.g. to simulate I/O errors.

Reply:
Kept for I/O error simulation.